tests/PHPStan/Type/CallableStringTypeTest.php

```diff
@@ -0,0 +1,110 @@
+<?php declare(strict_types = 1);
+
+namespace PHPStan\Type;
+
+use PHPStan\Testing\TestCase;
+use PHPStan\TrinaryLogic;
+use PHPStan\Type\Constant\ConstantStringType;
+
+class CallableStringTypeTest extends TestCase
+{
+
+	public function dataIsSuperTypeOf(): iterable
+	{
+		yield [
+			new CallableStringType(),
+			new UnionType([new ConstantStringType('test_function'), new ConstantStringType('test_function_2')]),
+			TrinaryLogic::createYes(),
+		];
+
+		yield [
+			new CallableStringType(),
+			new UnionType([new ConstantStringType('test_function'), new StringType()]),
+			TrinaryLogic::createMaybe(),
+		];
+
+		yield [
+			new CallableStringType(),
+			new UnionType([new IntegerType(), new FloatType()]),
+			TrinaryLogic::createNo(),
+		];
+
+		yield [
+			new CallableStringType(),
+			new ConstantStringType('test_function'),
+			TrinaryLogic::createYes(),
+		];
+
+		yield [
+			new CallableStringType(),
+			new CallableStringType(),
+			TrinaryLogic::createYes(),
+		];
+
+		yield [
+			new CallableStringType(),
+			new StringType(),
+			TrinaryLogic::createMaybe(),
+		];
+	}
+
+	/**
+	 * @dataProvider dataIsSuperTypeOf
+	 */
+	public function testIsSuperTypeOf(CallableStringType $type, Type $otherType, TrinaryLogic $expectedResult): void
```

Comment:
You can also add some testcases to TypeCombinatorTest::dataIntersect. It should work like this:

* Intersection of string and callable-string -> callable-string.
* Intersection of ConstantStringType and CallableStringType - depends on the value, can be normalized to ConstantStringType, it might not. Please test both cases.